### ISSUE TITLE
How to resolve Unable to download meta file: https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta?

### ISSUE THREAD

**Issue by richa-p** [dependency-check/DependencyCheck#3682]
I have been facing this issue in both these cases - 

1. Used brew install to install the executable
2. Cloned the project on local, followed the command to build and use the executable from target folder. 

No matter which option I'm trying I get the following error. I have tried all the solutions provided on - https://github.com/jeremylong/DependencyCheck/issues/2222

[INFO] Checking for updates
[ERROR] Unable to download meta file: https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta
org.owasp.dependencycheck.data.update.exception.UpdateException: Unable to download meta file: https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.getMetaFile(NvdCveUpdater.java:350)
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.getUpdatesNeeded(NvdCveUpdater.java:405)
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.update(NvdCveUpdater.java:122)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:873)
	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:680)
	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:606)
	at org.owasp.dependencycheck.App.runScan(App.java:256)
	at org.owasp.dependencycheck.App.run(App.java:188)
	at org.owasp.dependencycheck.App.main(App.java:83)
Caused by: org.owasp.dependencycheck.utils.DownloadFailedException: Download failed, unable to retrieve 'https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta'; Error downloading file https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta; unable to connect.
	at org.owasp.dependencycheck.utils.Downloader.fetchContent(Downloader.java:187)
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.getMetaFile(NvdCveUpdater.java:342)
	... 8 common frames omitted
Caused by: org.owasp.dependencycheck.utils.DownloadFailedException: Error downloading file https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta; unable to connect.
	at org.owasp.dependencycheck.utils.HttpResourceConnection.obtainConnection(HttpResourceConnection.java:267)
	at org.owasp.dependencycheck.utils.HttpResourceConnection.fetch(HttpResourceConnection.java:163)
	at org.owasp.dependencycheck.utils.Downloader.fetchContent(Downloader.java:182)
	... 9 common frames omitted
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:324)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:267)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:262)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1331)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1206)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1153)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:422)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:181)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:164)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1460)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1368)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:437)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:567)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:171)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:142)
	at org.owasp.dependencycheck.utils.HttpResourceConnection.obtainConnection(HttpResourceConnection.java:206)
	... 11 common frames omitted
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:384)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:289)
	at java.base/sun.security.validator.Validator.validate(Validator.java:264)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:231)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:132)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1315)
	... 25 common frames omitted
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
	at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:379)
	... 30 common frames omitted
[WARN] Unable to update 1 or more Cached Web DataSource, using local data instead. Results may not include recent vulnerabilities.
[ERROR] Unable to continue dependency-check analysis.
[ERROR] One or more fatal errors occurred
[ERROR] Unable to download meta file: https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2003.meta
[ERROR] No documents exist

**Comment by mattmook** [comment]
I'm getting a similar issue with using the gradle plugin, fails when running from my Mac but weirdly works when run via GitHub Actions.

**Comment by mprins** [comment]
based on the very limited information you have provided the conclusion is that something is off with your Java stack as it fails to do the certificate validation required for HTTPS

> Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

this is a runtime error.
The cause of this may be an intermediate "Let's Encrypt" certificate that has expired, see https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/ and https://community.letsencrypt.org/t/help-thread-for-dst-root-ca-x3-expiration-september-2021/149190
you may be able to fix it removing the expired certificate of updating your Java runtime

**Comment by jeremylong** [comment]
There has been another download issue (https://github.com/jeremylong/DependencyCheck/issues/3502) - however, as this is an SSL Handshake issue - likely due to the let's encrypt cert as Mark has pointed out.

**Comment by jeremylong** [comment]
Depending on the exact cause of the error - this could be resolved with 6.4.1. Alternatively, if you are still seeing this issue please see https://github.com/jeremylong/DependencyCheck/issues/728#issuecomment-301240883

**Comment by jaydubb12** [comment]
The easiest way I have found to solve the issue is as follows...hope this helps some folks in the future
1. Navigate to the parent domain for the CVE meta data `https://nvd.nist.gov/`
2. Use your browser to download the https certificate, which will be named `*.nist.gov.cer`
3. Install the certificate in your java keystore using the command line...here's an example`keytool -import -alias <yourAliasName> -file <pathToCerFile> -cacerts`

Once these steps are completed, the file will be stored in your keystore.  For some users, you may also have to install the cert in your `Keychain` (i.e. some MacOS users) as well...but I personally did not have to in order for this to work.

Also note....

1. If you have multiple Java versions installed, you should ONLY need to install this in the version you are using to build / run the java application

2. Simply installing this in the IDE certificate store will NOT be enough

**Comment by zieka** [comment]
Hi @eps1lon,
Thanks for the great feedback.  I agree with your points and don't think any piece is missing from your codemod (at least not that I saw).  I opted for making utility classes like this for my own projects because the `PropsWithChildren`, while awesome and useful, was too verbose for my liking (just a personal preference). 

I understand having nested generics makes it more flexible and agree `PropsWithChildren` is the more powerful API but the original intent was not to provide a more powerful API but to provide a drop-in replacement for `FunctionComponent` that required no change to imports.

If we don't feel such a drop-in replacement is desirable then it would make more sense for this code to live in a separate package.
